repositories/src/main/scala/cromwell/pipeline/datastorage/dto/ProjectSearch.scala

```diff
@@ -0,0 +1,68 @@
+package cromwell.pipeline.datastorage.dto
+
+import play.api.libs.json._
+
+sealed trait ProjectSearchFilter
+object ProjectSearchFilter {
+  implicit lazy val projectSearchFilterReads: Reads[ProjectSearchFilter] = Reads[ProjectSearchFilter] { json =>
+    (json \ "type").validate[String].flatMap {
+      case "all"            => json.validate[AllFilter.type]
+      case "name"           => json.validate[ByNameFilter]
```

Comment:
I think we can be better then this.
Play Json has support for custom naming strategy and using discriminator field which can help with automatic deserilization of sealed trait.
https://www.playframework.com/documentation/2.8.x/ScalaJsonAutomated
Also, take a look at `cromwell.pipeline.utils.json.AdtJsonFormatter` and its usages in `cromwell.pipeline.datastorage.dto.TypedValue#typedValueFormat`. This solves the same issue in a different (maybe more handy in this case) way.
So, please use existing tools.